### PR TITLE
fix(rds): unable to use tokens as port in `DatabaseInstance`

### LINKED ISSUE #17948: (aws-rds): Unable to use CfnParameter as port
### What is the problem?

CfnParameters cannot be used for the port value of a DatabaseInstance in aws-rds. The value requires only a number and valueAsNumber on a CfnParameter doesn't produce a valid number on synthesis.
`"Port": "-1.8881545897088186e+289"`
instead of
`"ToPort": {
          "Ref": "DBPort"
        }` (as it becomes for a security group)
https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_rds.DatabaseInstance.html

### Reproduction Steps

```ts
const dbPort = new CfnParameter(this, 'DBPort', {
  type: 'Number',
  default: 5432
})

const postgresDb = new rds.DatabaseInstance(this, 'PostgresDb', {
      port: dbPort.valueAsNumber,
})
```

### What did you expect to happen?

Expected that the generated cfn-template would have the port parameter as a reference to the cfn parameter

### What actually happened?

The port parameter was set to a static value: -1.8881545897088186e+289

### CDK CLI Version

2.0.0 (build 4b6ce31)

### Framework Version

_No response_

### Node.js Version

v16.9.1

### OS

mac os

### Language

Typescript

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by jumic**
In `DatabaseInstance` the port number can be specified using data type number.
If a token value (e.g. CloudFormation Parameter) was used here, the token was not resolved correctly.

The conversion of property `port` has been corrected by using method `Tokenization.stringifyNumber()`.

Fixes #17948.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-rds/test/instance.test.ts [hunk 1]
```diff
@@ -1630,6 +1630,38 @@ describe('instance', () => {
       },
     });
   });
+
+  test('instance with port', () => {
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/test/instance.test.ts:1634]:
```suggestion
  test('instance with port provided as a number', () => {
```

File: packages/@aws-cdk/aws-rds/lib/instance.ts [hunk 2]
```diff
@@ -746,7 +746,7 @@ abstract class DatabaseInstanceNew extends DatabaseInstanceBase implements IData
       performanceInsightsRetentionPeriod: enablePerformanceInsights
         ? (props.performanceInsightRetention || PerformanceInsightRetention.DEFAULT)
         : undefined,
-      port: props.port?.toString(),
+      port: props.port ? Tokenization.stringifyNumber(props.port) : undefined,
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/lib/instance.ts:749]: Two things:

1. I think you want the condition to be `props.port === undefined`, as 0 is falsy in JavaScript.
2. I believe you can use `Token.toString(props.port)` instead - it's more of the idiomatic way to do it (`Tokenization` is more of an internal class).

File: packages/@aws-cdk/aws-rds/test/instance.test.ts [hunk 3]
```diff
@@ -1630,6 +1630,38 @@ describe('instance', () => {
...
+      port: 3306,
+    });
+
+    // THEN
+    expect(stack).toHaveResourceLike('AWS::RDS::DBInstance', {
+      Port: '3306',
+    });
+  });
+
+  test('instance with port from token', () => {
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/test/instance.test.ts:1649]:
```suggestion
  test('instance with port provided as a CloudFormation parameter', () => {
``

File: packages/@aws-cdk/aws-rds/test/instance.test.ts [hunk 4]
```diff
@@ -1630,6 +1630,38 @@ describe('instance', () => {
...
+    // WHEN
+    new rds.DatabaseInstance(stack, 'Database', {
+      engine: rds.DatabaseInstanceEngine.MYSQL,
+      instanceType: ec2.InstanceType.of(ec2.InstanceClass.BURSTABLE2, ec2.InstanceSize.SMALL),
+      vpc,
+      port: cdk.Token.asNumber({ Ref: 'port' }),
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/test/instance.test.ts:1655]: Let's use a `CfnParameter` instead (the test will be more realistic this way).

**Review by skinny85 — CHANGES_REQUESTED** [review]
Thanks for the contribution @jumic! A few tiny comments first.

File: packages/@aws-cdk/aws-rds/lib/instance.ts (hunk 2 above)
Comment (jumic) [reply packages/@aws-cdk/aws-rds/lib/instance.ts]: 1. Good catch. 0 is a valid port number. I changed it.
2. Method `Token.toString()` doesn't accept an attribute. So I can't use it. Should I try something else? Or is `Tokenization.stringifyNumber(props.port)` the only way to handle it?

**Review by skinny85 — APPROVED** [review]
Awesome, thanks for the contribution @jumic!

**Pull request merged**